compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -310,21 +310,22 @@ OMR::Z::TreeEvaluator::dsqrtEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
    TR::Node * firstChild = node->getFirstChild();
    TR::Register * targetRegister = NULL;
-   TR::Register * opRegister = cg->evaluate(firstChild);
 
-   if (cg->canClobberNodesRegister(firstChild))
+   //See whether to use SQDB or SQDBR depending on how many times it is referenced
+   if (firstChild->isSingleRefUnevaluated() && firstChild->getOpCodeValue() == TR::dloadi)
       {
-      targetRegister = opRegister;
+      targetRegister = cg->allocateRegister(TR_FPR);
```

Comment:
This variable is initialized in the same way on both paths. Can we common this outside of the if?